File: Trinity.Runtime/platform_src/ios/elastOS/Classes/ContactNotifier/comm/SetPresenceCommand.swift
import ElastosCarrierSDK

public class SetPresenceCommand : CarrierCommand {
    private let helper: CarrierHelper
    private let status: CarrierPresenceStatus

    init(helper: CarrierHelper, status: CarrierPresenceStatus) {
        self.helper = helper
        self.status = status
    }

    public func executeCommand() {
        Log.i(ContactNotifier.LOG_TAG, "Executing presence status command")
        do {
            try helper.carrierInstance!.setSelfPresence(status)
        }
        catch (let error) {
            print(error)
        }
    }
}

File: Trinity.Runtime/platform_src/android/app/src/main/java/org/elastos/trinity/runtime/contactnotifier/comm/SetPresenceCommand.java
package org.elastos.trinity.runtime.contactnotifier.comm;

import android.util.Log;

import org.elastos.carrier.PresenceStatus;
import org.elastos.trinity.runtime.contactnotifier.ContactNotifier;
import org.elastos.trinity.runtime.contactnotifier.RemoteNotificationRequest;
import org.json.JSONObject;

public class SetPresenceCommand implements CarrierCommand {
    private CarrierHelper helper;
    private PresenceStatus status;

    SetPresenceCommand(CarrierHelper helper, PresenceStatus status) {
        this.helper = helper;
        this.status = status;
    }

    @Override
    public void executeCommand() {
        Log.i(ContactNotifier.LOG_TAG, "Executing presence status command");
        try {
            helper.carrierInstance.setPresence(status);
        }
        catch (Exception e) {
            e.printStackTrace();
        }
    }
}